File: LeetCode/Easy/0344-reverse-string/0344-reverse-string.cs
public class Solution {
    public void ReverseString(char[] s) {
        var left = 0;
        var right = s.Length - 1;
        while (left < right)
        {
            // could use tuple swap, however it could be slightly slower
            var tmp = s[left];
            s[left] = s[right];
            s[right] = tmp;

            left++;
            right--;
        }
    }
}

File: LeetCode/Easy/0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        right = -1

        for i in range(len(s) // 2):
            s[i], s[right] = s[right], s[i]
            right -= 1